src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSender.java

```diff
@@ -49,21 +50,27 @@
             .register();
 
     protected final String th2Pin;
+    private final AtomicReference<ConnectionManager> connectionManager = new AtomicReference<>();
     private final AtomicReference<String> routingKey = new AtomicReference<>();
     private final AtomicReference<String> exchangeName = new AtomicReference<>();
-    private final AtomicReference<ConnectionManager> connectionManager = new AtomicReference<>();
+    private final String queueName;
+    private final long sizeLimit;
+    private final int recheckSizeTimeoutSeconds;
     private final String th2Type;
 
     public AbstractRabbitSender(
             @NotNull ConnectionManager connectionManager,
-            @NotNull String exchangeName,
-            @NotNull String routingKey,
+            @NotNull QueueConfiguration queueConfiguration,
             @NotNull String th2Pin,
             @NotNull String th2Type
     ) {
         this.connectionManager.set(requireNonNull(connectionManager, "Connection can not be null"));
-        this.exchangeName.set(requireNonNull(exchangeName, "Exchange name can not be null"));
-        this.routingKey.set(requireNonNull(routingKey, "Routing key can not be null"));
+        requireNonNull(queueConfiguration, "Queue configuration can not be null");
+        exchangeName.set(requireNonNull(queueConfiguration.getExchange(), "Exchange name can not be null"));
+        routingKey.set(requireNonNull(queueConfiguration.getRoutingKey(), "Routing key can not be null"));
+        queueName = requireNonNull(queueConfiguration.getQueue(), "Queue name can not be null");
```

Comment:
Destination queues are dynamic and depend on the current schema. Please note: infra components can change bindings between routing keys and queues without box restarting in RabbitMQ.
We should calculate actual set of destination ques before every check

Reply:
Do you mean that `QueueConfiguration.queue` from `mq.json` may change? I thought that is configuring once before start.

Reply:
A pin with `publish` attributes has `QueueConfiguration.routingKey` and hasn't got `QueueConfiguration.queue`.
Binding between the routing kay and queue(s) is declared in the RabbitMQ and our box doesn't know about the current binding.